lib/vim-state.coffee

```diff
@@ -661,15 +650,26 @@ class VimState
     text = @getRegister(name)?.text
     @editor.insertText(text) if text?
 
-  ensureCursorIsWithinLine: (cursor) =>
-    return if @processing or @mode isnt 'normal'
+  # Private: ensure the mode follows the state of selections
+  checkSelections: =>
+    return unless @editor?
+    if @editor.getSelections().every((selection) -> selection.isEmpty())
+      @ensureCursorsWithinLine() if @mode is 'normal'
```

Comment:
It looks like you can remove the `if @mode is 'normal'` here, since `::ensureCursorsWithinLine` internally guards that the mode is normal.